x-pack/plugins/spaces/public/types.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { GetSpaceResult } from '../common';
+
+export interface SpacesData {
+  readonly spacesMap: Map<string, SpaceData>;
+  readonly activeSpaceId: string;
+}
+
+export interface SpaceData extends Omit<GetSpaceResult, 'disabledFeatures'> {
```

Comment:
I agree, I was't sure what to name it tbh, because I was thinking the SpacesContext could be used for other components in the future, such as `CopySavedObjectsToSpaceFlyout`. I suppose we can always change the name of this type later if/when we decide to do that though. I'll go ahead and change it to `ShareToSpaceTarget` in this PR.